lib/prometheus/engineering_collector.rb

```diff
@@ -1,12 +1,17 @@
 # frozen_string_literal: true
 
+require 'lru_redux'
+
 unless defined? Rails
   require File.expand_path('../../config/environment', __dir__)
 end
 
 # when /metrics is called. This runs directly on prometheus_exporter.
 class EngineeringCollector < PrometheusExporter::Server::TypeCollector
+  LRU_DURATION = 1.hour
+
   def initialize
+    @cache = LruRedux::TTL::ThreadSafeCache.new(1, LRU_DURATION.seconds.to_i)
```

Comment:
Why introduce a new dependency when we could use redis with Rails.cache?

Reply:
Can we specify timout with generic Rails caching?

Reply:
If not, we can rely on redis' own methods, there you can set a TTL.

Reply:
Yes, there is an `expires_in` option. See https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch

Reply:
I am opposed to adding a new unnecessary dependency. Looking at the gem, it doesn't even appear active (last changes were in 2017; last release was March 2015) or well documented (docs link broken: https://www.rubydoc.info/gems/lru_redux/1.1.0). We need to be careful when selecting new dependencies and ideally not introduce them at all where not necessary.

Reply:
Yes, I'm on the same page. The lib was recommended by Racecar. I've checked activeness of that project and it seemed fine (code is stable and there are discussions happening).